File: python/euler24.py
#!/usr/bin/env python3

from math import factorial

def permute(iterable):
    iterable = sorted(iterable)
    n = len(iterable)
    nperms = factorial(n)
    count = 0
    while count < nperms:
        yield iterable
        j, k = 0, 0
        for i in range(n - 1):
            if iterable[i] < iterable[i + 1]:
                k = i
        for i in range(k, n):
            if iterable[k] < iterable[i]:
                j = i
        iterable[j], iterable[k] = iterable[k], iterable[j]
        iterable[k + 1:] = iterable[k + 1:][::-1]
        count += 1

def main():
    count = 0
    for i in permute([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]):
        count += 1
        if count == 1000000:
            print(i)
            break

if __name__ == '__main__':
    main()

File: src/euler24.c
#include <stdio.h>
// #include <math.h>

unsigned factorial(unsigned n) {
    unsigned fac = 1;
    while (n > 1) {
        fac *= n;
        --n;
    }
    return fac;
}

int main() {
    int iterable[3] = {0, 1, 2};
    unsigned nperms = factorial(3);
    int permutations[3][nperms];
    for (int i = 0; i <)
    printf("%d", nperms);
    return 0;
}
